### ISSUE TITLE
on-search-change event is not that useful

### ISSUE THREAD

**Issue by asvae** [monterail/vue-multiselect#48]
As I'm using `on-search-change` event, the main intention is to load options from the server. But it doesn't suit the task. Here's why. 

`on-search-change` event triggers:
1) Page ready.
2) Parent `selected` prop externally changed.
3) User clicked on component (triggers 2 times).
4) User types any letter

`load-options` event should trigger:
1) User clicked on component (once).
2) User types any letter

This is okeyish when you have one select on page. And not so much with ton of forms.

Hence, I humbly propose to deprecate `on-search-change` and implement `load-options`. And I'm willing to shoulder the task if help is wanted.

**Comment by shentao** [comment]
I want to deprecate some more of the callback functions (together with support for `.sync`) that we have right now and on-search-change is one of them.
Instead, we could simply `$emit` events like `on-change`, `on-search-change` etc. This would probably make it easier to control and would simplify parts of the code.
I will try to implement it this weekend.

**Comment by dobromir-hristov** [comment]
Just what I was looking for, there are no events emitted atm. A on-closed event would be nice, if you want to use the multiple multiselect`s selected data just after some is done with their selecting?

**Comment by shentao** [comment]
Hey @asvae, @dobromir-hristov!

Please look into #57 as this might be something you had in mind.

**Comment by dobromir-hristov** [comment]
Yeah I saw. Will it support current vue 1.x ?

**Comment by shentao** [comment]
So far yes. It just requires the developer to always react to the `change` event to change the `selected` model.

**Comment by dobromir-hristov** [comment]
Just a quick one. How do I know from where the change event is fired? I cant pass the event it self to my method, right?
I mean I dont want to have a method for every single multiselect, because most of them are generated within a foreach.

**Comment by shentao** [comment]
Been thinking about this lately.
Came up with an idea of a unique ID of the component, that would be always passed as a second argument on every event from #57. But this would be only possible after those changes are published.

Right now I avoid this problem in my project with `v-for` looping a smaller component (per record) where each has one multiselect. Thus I can use a single component’s method. This actually makes sense for me, as I also have other inputs that I have to loop for each record.

**Comment by dobromir-hristov** [comment]
Yeah, and ID of some sort would be good. It can even be a prop that gets passed from the parent. Well I will just have to go with what I got atm. The emit event is pretty much the same as the onChange prop from before so, I will probably roll back to the master branch.

**Comment by shentao** [comment]
The PR is work in progress so I wouldn’t use it in production. I still have to fix the tests and do some more changes. Also as this would deprecate the `.sync` option (major breaking change) I need to be 100% sure it works fine. Yes, the ID would be passed as a prop. Creating unique ID inside the component is out of the responsibility of the component.
I would try the approach with forEaching a small, wrapper component where you can put multiselect.

**Comment by asvae** [comment]
@dobromir-hristov 
> How do I know from where the change event is fired?

Like this, I guess?
Template:
```html
<vm-component
    @on-change="process"
></vm-component>
```

Script -> methods:
```es6
process (event){
    // do stuff with event
}
```

And you also can
```html
<vm-component
    @on-change="process($arguments[0])"
></vm-component>
```
and pass the data object from the select

**Comment by dobromir-hristov** [comment]
Hm did not know if I could use $arguments. Could try later.

**Comment by shentao** [comment]
@dobromir-hristov @asvae Fixed in v1.0.0
